src/main/java/ca/corefacility/bioinformatics/irida/service/impl/ProjectServiceImpl.java

```diff
@@ -707,6 +701,11 @@ public void removeReferenceFileFromProject(Project project, ReferenceFile file)
 	@PreAuthorize("hasRole('ROLE_ADMIN') or hasPermission(#project, 'canManageLocalProjectSettings')")
 	public Join<Project, UserGroup> addUserGroupToProject(final Project project, final UserGroup userGroup,
 			final ProjectRole role) {
+		Set<UserGroupJoin> userGroupJoins = userGroup.getUsers();
```

Comment:
Instead of calling `getUsers` and making the collection eager loading in the UserGroup can you change this to call `userGroupJoinRepository.findUsersInGroup(userGroup)`

Reply:
Please see aa963b11bf9e67e6e18f08d357096ded36438945 and